src/bp/core/services/hook/hook-service.ts

```diff
@@ -277,6 +282,26 @@ export class HookService {
     hook.debug.forBot(botId, 'after execute')
   }
 
+  private addEventStep = (hookName: string, status: 'completed' | 'error', hook: Hooks.BaseHook, error?: any) => {
+    if (!hook.args?.event) {
+      return
+    }
+
+    const event = hook.args.event as sdk.IO.Event
+    if (error) {
+      addErrorToEvent(
+        {
+          type: 'hook-execution',
+          stacktrace: error.stacktrace || error.stack,
+          actionArgs: _.omit(hook.args, ['bp', 'event'])
+        },
+        event
+      )
+    }
+
+    addStepToEvent(`hook:${hookName}:${status}`, event)
```

Comment:
Do we need to remember that naming logic in order for it to be displayed /processed properly  ? 

if yes I hate it

Reply:
The goal was to store something compact for the event which could be parsed (type:name:status), otherwise it's a big object, which is not really necessary